### PR TITLE
Pull Request for Issue2400: Minor tweaks to BioXAS beamline status

### PR DESCRIPTION
**Pull request by helfrij**
- Added `buttonClicked(...)` and `controlClicked(...)` signals to the CLSButtonBar and CLSControlButtonBar classes. This allows the view to announce when controls have been clicked, instead of just when the selection state changes.
- Added signal to the BioXASPersistentView to announce when a beam status bar control is clicked.
- Modified BioXASAppController to switch the current pane to the beam status pane in response to the persistent view's new signal.

### CODE REVIEW

File: source/application/CLS/CLSAppController.cpp [hunk 1]
```diff
@@ -119,19 +119,18 @@ void CLSAppController::setupUserInterfaceImplementation()
...
 }
 
-void CLSAppController::addMainWindowViewToPane(QWidget *view, const QString &viewName, const QString &paneCategoryName, const QString &paneIcon)
+void CLSAppController::addMainWindowView(QWidget *view, const QString &viewName, const QString &paneCategoryName, const QString &paneIcon)
 {
 	if (view) {
-		QWidget *mainWindowView = AMMainWindow::buildMainWindowPane(viewName, paneIcon, view);
-		addViewToPane(mainWindowView, viewName, paneCategoryName, paneIcon);
+		QWidget *pane = AMMainWindow::buildMainWindowPane(viewName, paneIcon, view);
+		viewPaneMapping_.insert(view, pane);
```
Comment (XuetaoLiu-CLS) [inline source/application/CLS/CLSAppController.cpp]: @helfrij Thanks for changing this :)

Comment (helfrij) [inline source/application/CLS/CLSAppController.cpp]: 😄

**Comment by helfrij** [comment]
Hey @XuetaoLiu-CLS, would you review?

File: source/application/BioXAS/BioXASAppController.cpp [hunk 2]
```diff
@@ -362,61 +362,61 @@ void BioXASAppController::createPersistentView()
...
+	addMainWindowView( createComponentView(bioXASBL->beWindow()), "Be Window", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->endstationTable()), "Endstation Table", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->dbhrMirrors()), "DBHR Mirrors", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->jjSlits()), "JJ Slits", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->xiaFilters()), "XIA Filters", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->standardsWheel()), "Standards Wheel", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->cryostatStage()), "Cryostat Stage", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->filterFlipper()), "Filter Flipper", componentPaneCategoryName_, componentPaneIcon_);
+	addMainWindowView( createComponentView(bioXASBL->sollerSlit()), "Soller slits", componentPaneCategoryName_, componentPaneIcon_);
```
Comment (XuetaoLiu-CLS) [inline source/application/BioXAS/BioXASAppController.cpp]: A nitpicking, the names of the other components is like camel case. Do you want `Soller slits` to be `Soller Slits`?

File: source/ui/CLS/CLSBeamlineStatusView.h [hunk 3]
```diff
@@ -31,6 +31,8 @@ class CLSBeamlineStatusView : public QWidget
 	void beamStatusChanged(CLSBeamlineStatus *newStatus);
 	/// Notifier that the selected component has changed.
 	void selectedComponentChanged(AMControl *newControl);
```
Comment (XuetaoLiu-CLS) [inline source/ui/CLS/CLSBeamlineStatusView.h]: I think this signal should the the same as the new added signal `controlClicked()`. I think we need only one.

Comment (XuetaoLiu-CLS) [inline source/ui/CLS/CLSBeamlineStatusView.h]: I see... they are used in different scenario.

**Comment by XuetaoLiu-CLS** [comment]
Looks good.

Time Estimate: 5 minutes

**Comment by dretrex** [comment]
Looks good.  There are some merge conflicts that need to be resolved, but then it can be merged!
Time Estimate: 3 minutes

File: source/application/BioXAS/BioXASSideAppController.cpp [hunk 4]
```diff
@@ -41,10 +41,10 @@ void BioXASSideAppController::updateGeDetectorView()
     QWidget *detectorPane = viewPaneMapping_.value(detectorView, 0);
 
     if (detector && detectorView && detectorPane) {
-	if (detector->isConnected())
-	    mw_->showPane(detectorPane);
-	else
-	    mw_->hidePane(detectorPane);
+		if (detector->isConnected())
+			mw_->showPane(detectorPane);
```
Comment (dretrex) [inline source/application/BioXAS/BioXASSideAppController.cpp]: Maximum tabbage!

**Comment by helfrij** [comment]
Merge conflicts should be resolved. Thanks @XuetaoLiu-CLS @dretrex!

**Pull request merged**